app/contracts/delete_contract.rb

```diff
@@ -59,8 +59,10 @@ def authorized?
       user.admin? && user.active?
     when Proc
       instance_exec(&permission)
+    when Symbol
+      model.project && user.allowed_in_project?(permission, model.project)
     else
-      !model.project || user.allowed_in_project?(permission, model.project)
+      raise ArgumentError, "You need to set a delete_permission Proc, or project-based permission"
```

Comment:
Should we add a custom exception to give more context, `ContractError::PermissionNotSet` and the message with the name of the contract?